Functions/xASL_io_ReadDataPar.m

```diff
@@ -140,36 +140,167 @@
 	x = xASL_io_ReadDataPar_FixFields(x);
 	
 	%% Check deprecated fields
-	if isfield(x,'SegmentSPM12')
-		warning('Deprecated field. Please use x.settings.SegmentSPM12 instead of x.SegmentSPM12');
-		if ~isfield(x,'settings') || ~isfield(x.settings,'SegmentSPM12')
-			x.settings.SegmentSPM12 = x.SegmentSPM12;
-		end
-		x = rmfield(x,'SegmentSPM12');
-	end
-	
-	if isfield(x,'M0_conventionalProcessing')
-		warning('Deprecated field. Please use x.settings.M0_conventionalProcessing instead of x.M0_conventionalProcessing');
-		if ~isfield(x,'settings') || ~isfield(x.settings,'M0_conventionalProcessing')
-			x.settings.M0_conventionalProcessing = x.M0_conventionalProcessing;
-		end
-		x = rmfield(x,'M0_conventionalProcessing');
-	end
+    x = xASL_io_ReadDataParCheckDeprecatedFields(x);
 	
-	if isfield(x,'exclusion')
-		warning('Deprecated field. Please use x.dataset.exclusion instead of x.exclusion');
-		if ~isfield(x,'dataset') || ~isfield(x.dataset,'exclusion')
-			x.dataset.exclusion = x.exclusion;
-		end
-		x = rmfield(x,'exclusion');
-	end
 	
 else
 	error('Unknown file extension');
 end
 
 end
 
+
```

Comment:
The example at lines 32-38 has to be updated

Reply:
6b34e0a